WT-10027, WT-8848, WT-9879 v5.0 group backport

ruby-oujo opened the pull request:
****Note I am following the changes we made with v6.0 where we are not changing the eviction flags for rollback and correcting the flag names that was forgotten in one of the changes.** 

[WT-8848](https://jira.mongodb.org/browse/WT-8848) [Add session statistics that counts a transaction's dirty bytes.](https://github.com/wiredtiger/wiredtiger/pull/8915/commits/9dd76f02373e9858c6a9e8456027bd15d25e967f) 

* Add session statistics that counts a transaction's dirty bytes.
* Add test for session transaction dirty bytes statistic.

(cherry picked from commit https://github.com/wiredtiger/wiredtiger/commit/3112dccffa27bb8152bbc8ed22f63a0361cf28d7)

[WT-9879](https://jira.mongodb.org/browse/WT-9879) [Fix overcounting of session txn dirty bytes statistic ](https://github.com/wiredtiger/wiredtiger/pull/8915/commits/e25f57f40e433c310d040bf321a8feab69a9c472)https://github.com/wiredtiger/wiredtiger/pull/8319)
(cherry picked from commit https://github.com/wiredtiger/wiredtiger/commit/20386a28e2b57b1f8d2f4c3d8b894180876911db)


[WT-10027](https://jira.mongodb.org/browse/WT-10027) [Fix session txn statistic to exclude eviction work. ](https://github.com/wiredtiger/wiredtiger/pull/8915/commits/374ef2e2d8d5feca99cb1ab12c6e209b3de89bf1)
(cherry picked from commit https://github.com/wiredtiger/wiredtiger/commit/d5edce423f92ef3208b711735b803cc712ff807b)

ruby-oujo commented:
I am not sure why test_stat08 is failing, but we haven't got a backport open for a change made in WT-9774 for test_stat08 that was backported with this group for v6.0.

ruby-oujo commented:
I have tested with the WT-9774 changes, however we do not have the debug configuration available and we still fail the test on ubuntu, however the windows failure disappears. Backporting WT-9774 doesn't appear to fix this bug.

It is interesting the current patch is failing on windows, because the area of code where it is failing says we will skip the time check on windows - but the check continues. 

This failure looks like a testing bug, or could be a legitimate error but I am uncertain as the error message is quite vague and I am not familiar with the test code @sueloverso could you take a look at the two failures and see if this is a test issue?

sueloverso commented:
@ruby-oujo the failures are occurring in a part of the test that is not related to the txn dirty bytes change. Maybe the rollback instead of a commit is affecting something? The backport looks clean to me.

sulabhM commented:
@ruby-oujo , please file a bug report for the existing bugs.

ruby-oujo commented:
> @ruby-oujo the failures are occurring in a part of the test that is not related to the txn dirty bytes change. Maybe the rollback instead of a commit is affecting something? The backport looks clean to me.

Thanks @sueloverso , I've changed the rollbacks to commits and added a cursor debug flag as @kommiharibabu suggests.

kommiharibabu reviewed (approved):
LGTM

sulabhM commented on src/include/btree_inline.h:
```diff
@@ -225,6 +225,16 @@ __wt_cache_page_inmem_incr(WT_SESSION_IMPL *session, WT_PAGE *page, size_t size)
+        /*
+         * For application threads, track the transaction bytes added to cache usage. We want to
+         * capture only the application's own changes to page data structures. Exclude changes to
+         * internal pages or changes that are the result of the application thread being co-opted
```
I have created WT-10788. Intuitively internal pages should be included if the application thread needs to do an eviction to estimate how much dirty content comes from a particular transaction. But Josef ran many tests and wrote a document we used to fine-tune. I don't remember now what we thought about the internal pages. We will have to revisit the work done during that time.
